src/uct/cuda/cuda_copy/cuda_copy_md.c

```diff
@@ -689,6 +689,13 @@ uct_cuda_copy_md_query_attributes(uct_cuda_copy_md_t *md, const void *address,
         return UCS_ERR_INVALID_ADDR;
     }
 
+    if ((uintptr_t)base_address + alloc_length < (uintptr_t)address + length) {
+        /* >1 PA behind VA range, treat as managed memory to force pipeline */
+        mem_info->type = UCS_MEMORY_TYPE_CUDA_MANAGED;
+        base_address   = (CUdeviceptr)address;
+        alloc_length   = length;
+    }
+
```

Comment:
Is `cuMemGetAddressRange` support for VMM allocations documented somewhere? The driver API docs mention legacy allocators only.
`Returns the base address in *pbase and size in *psize of the allocation by cuMemAlloc() or cuMemAllocPitch()`.

Reply:
Indeed not well documented. The behavior we've seen is that it is able to provide base pointer and size corresponding to base address and length of mapped address space from the corresponding cuMemMap.